src/app/topbar-reminders/topbar-reminders.component.ts

```diff
@@ -0,0 +1,25 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { Subscription } from 'rxjs';
+import { TopbarService } from '../shared/services/topbar.service';
+
+@Component({
+  selector: 'app-topbar-reminders',
+  templateUrl: './topbar-reminders.component.html',
+  styleUrls: ['./topbar-reminders.component.css']
+})
+export class TopbarRemindersComponent implements OnInit, OnDestroy {
+  isWalletInit = true;
+  walletSub: Subscription;
+
+  constructor(private topbarService: TopbarService) {
+    this.walletSub = this.topbarService.getWalletState().subscribe(res => {
+      this.isWalletInit = res;
+    });
+  }
+
+  ngOnInit() {}
+
+  ngOnDestroy() {
+    this.walletSub.unsubscribe();
+  }
+}
```

Comment:
You are on a good path, but the idea is a little bit different than what you've made.

Firstly, there is no need for `TopbarService` it shouldn't have a subject that the component subscribes to. Everything that `TopbarRemindersComponent` needs is to subscribe to `WalletService`'s and `UserService`'s observables which will tell when wallet and user data changes.

So, let's clarify things for `WalletService`. You need to make something like this:

```typescript
@Injectable()
export class WalletService {
    walletChange$: Observable<WalletDetails | null>;

    private walletChangeSubject = new ReplaySubject<WalletDetails | null>(1);

    constructor(private http: BackendHttpClient) {
        this.walletChange$ = this.walletChangeSubject.asObservable();
    }

    initWallet(address: string) {
        return this.http.post<WalletDetails>('/api/wallet/wallet',
            <InitWalletData>{
                public_key: address
            }).pipe(this.tapWalletChange);
    }

    getUserWallet() {
        return this.http.get<WalletDetails>('/api/wallet/wallet')
            .pipe(this.tapWalletChange);
    }

    private tapWalletChange(source: Observable<WalletDetails>) {
        return source.pipe(
            tap(wallet => {
                UserStatusStorage.walletData = wallet;
                this.walletChangeSubject.next(wallet);
            },
                _ => this.walletChangeSubject.next(null)
            )
        )
    }
}
```

Now, inside `TopbarRemindersComponent`, inside `ngOnInit()` you only need to subscribe to `walletChange$` observable. If it receives `null`, set appropriate message to user, if it receives `WalletDetails`, don't show anything.

The same procedure applies when receiving user data from `UserService`, except, you will not need `null` value to return, you will always get user data, but you will need to check if user is verified or not.